app/controllers/purchases_controller.rb

```diff
@@ -9,86 +9,15 @@ def create
         return render json: { errors: [{ message: 'Cart not found!' }] }, status: :unprocessable_entity
       end
 
-      user = if cart.user.nil?
-               user_params = purchase_params[:user] ? purchase_params[:user] : {}
-               User.create(**user_params.merge(guest: true))
-             else
-               cart.user
-             end
+      user_params = purchase_params[:user] ? purchase_params[:user] : {}
```

Comment:
nesses casos em que o próprio valor usado no ternário vai ser inserido dentro da variável você pode utilizar o or short circuit

```suggestion
      user_params = purchase_params[:user] || {}
```